deps/src/main/java/com/microsoft/azure/sdk/iot/deps/transport/amqp/ErrorLoggingBaseHandler.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) Microsoft. All rights reserved.
+ *  Licensed under the MIT license. See LICENSE file in the project root for full license information.
+ */
+
+package com.microsoft.azure.sdk.iot.deps.transport.amqp;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.engine.BaseHandler;
+import org.apache.qpid.proton.engine.Event;
+
+@Slf4j
+public class ErrorLoggingBaseHandler extends BaseHandler
+{
+    protected ProtonJExceptionParser protonJExceptionParser;
+
+    @Override
+    public void onLinkRemoteClose(Event event)
+    {
+        protonJExceptionParser = new ProtonJExceptionParser(event);
+        if (protonJExceptionParser.getError() == null)
+        {
+            log.debug("Amqp link {} was closed remotely", event.getLink().getName());
+        }
+        else
+        {
+            if (event.getLink() != null && event.getLink().getName() != null)
+            {
+                log.warn("Amqp link {} was closed remotely with exception {} with description {}", event.getLink().getName(), protonJExceptionParser.getError(), protonJExceptionParser.getErrorDescription());
+            }
+            else
+            {
+                log.warn("Unknown amqp link was closed remotely with exception {} with description {}", protonJExceptionParser.getError(), protonJExceptionParser.getErrorDescription());
+            }
+        }
+    }
+
+    @Override
+    public void onSessionRemoteClose(Event event)
+    {
+        protonJExceptionParser = new ProtonJExceptionParser(event);
+        if (protonJExceptionParser.getError() == null)
```

Comment:
Do we allow multiple sessions over a single connection? If yes, then it'll be useful to log the session identifier here as well (similar to link).

Reply:
We do open multiple sessions for multiplexed connections, I believe. However, the proton-j session object doesn't have any information in it here that we could use to correlate back to which session it was. Sessions being closed remotely are a bit of a rare case, from my understanding, so we can hold off on adding that kind of mapping for now